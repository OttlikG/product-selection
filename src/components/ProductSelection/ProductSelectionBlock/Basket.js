
import React from 'react'

export default function Basket ({ data, push }) {
  function selectedProductionList (production) {
    return (
      <div key={production.id}>- { production.product_name}</div>
    )
  }

  return (
    <div className="block basket-block">
      <div>Basket</div>

      <div className="product-list">
        { data.map(selectedProductionList) }
      </div>

      <button className='checkout' onClick={() => push('confirmation')}>Checkout</button>
    </div>
  )
}
