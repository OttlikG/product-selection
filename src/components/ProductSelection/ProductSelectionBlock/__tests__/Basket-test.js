
import React from 'react'
import Basket from '../Basket'

describe('Basket', () => {
  let props, component
  const chelsea = {
    category: 'Sports',
    id: 'aaa',
    location: 'LONDON',
    product_name: 'Chelsea TV',
    isSelected: true
  }

  beforeEach(() => {
    props = {
      data: [chelsea],
      selectProductById: jest.fn(),
      push: jest.fn()
    }

    component = render(<Basket {...props} />)
  })

  it('should render 1 selected product', () => {
    expect(component.find('.product-list').children.length).toBe(1)
  })

  it('should jump to confirmation page when checkout clicked', () => {
    component.find('.checkout').props.onClick()
    expect(props.push).toHaveBeenCalledTimes(1)
    expect(props.push).toHaveBeenCalledWith('confirmation')
  })
})
