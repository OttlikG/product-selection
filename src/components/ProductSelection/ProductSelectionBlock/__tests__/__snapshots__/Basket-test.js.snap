// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket should jump to confirmation page when checkout clicked 1`] = `
<div
  className="block basket-block"
>
  <div>
    Basket
  </div>
  <div
    className="product-list"
  >
    <div>
      - 
      Chelsea TV
    </div>
  </div>
  <button
    className="checkout"
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;

exports[`Basket should render 1 selected product 1`] = `
<div
  className="block basket-block"
>
  <div>
    Basket
  </div>
  <div
    className="product-list"
  >
    <div>
      - 
      Chelsea TV
    </div>
  </div>
  <button
    className="checkout"
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;
